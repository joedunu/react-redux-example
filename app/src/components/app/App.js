// @flow

import React from 'react'
import { ConnectedRouter } from 'react-router-redux'
import { connect } from 'react-redux'
import { Route, Switch } from 'react-router-dom'

import HeaderContainer from '../common/header/HeaderContainer'
import SignUpContainer from '../signUp/SignUpContainer'
import UsersListContainer from '../usersList/UsersListContainer'
import Footer from '../common/footer/Footer'
import ReactModal from '../reactModal/ReactModal'
import EditUserContainer from '../editUser/EditUserContainer'

type AppProps = {
  store: Object,
  history: Object
}

const App = (props: AppProps) => {
  const history = props.history

  return (
    <div>
      <HeaderContainer history={history} />
      <ConnectedRouter history={history}>
        <Switch>
          <Route path='/sign-up' component={SignUpContainer} />
          <Route exact path='/' component={UsersListContainer} />
          <Route path='/react-modal' component={ReactModal} />
          <Route path='/edit-user' component={EditUserContainer} />
        </Switch>
      </ConnectedRouter>
      <Footer />
    </div>
  )
}

const AppContainer = connect(null, null)(App)
export default AppContainer
