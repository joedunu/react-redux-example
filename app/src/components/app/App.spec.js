import React from 'react'
import {shallow} from 'enzyme'
import {shallowToJson} from 'enzyme-to-json'
import configureMockStore from 'redux-mock-store'

import App from './App'

describe('App', () => {
  const mockStore = configureMockStore([])
  let store
  const history = {}
  let combinedReducersState = {
    form: {}
  }

  beforeEach(() => {
    store = mockStore(combinedReducersState)
  })

  test('should pass snapshot test', () => {
    const component = shallow(<App
      store={store}
      history={history}
    />)
    const tree = shallowToJson(component)
    expect(tree).toMatchSnapshot()
  })
})
