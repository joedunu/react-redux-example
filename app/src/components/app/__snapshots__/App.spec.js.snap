// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should pass snapshot test 1`] = `
<App
  dispatch={[Function]}
  history={Object {}}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
